packages/jest-worker/src/workers/ChildProcessWorker.ts

```diff
@@ -255,6 +255,19 @@ export default class ChildProcessWorker
   }
 
   private _onMessage(response: ParentMessage) {
+    // Ignore messages emitted by @esbuild-kit/esm-loader
+    // ref: https://github.com/esbuild-kit/esm-loader/issues/43
+    if (
+      typeof response === 'object' &&
+      'type' in response &&
+      typeof (response as any).type === 'string' &&
+      (response as any).type === 'dependency' &&
```

Comment:
Thanks for working on this PR @LinusU 

I think this check should enforce that it meets what jest expects, rather than specifically handling for tsx. e.g. that `response` is an array here, and remove the error on L327.

That way, it wouldn't fail with any other tools relying on IPC.

Reply:
Good thinking! Fixed ✅

Reply:
@privatenumber I've sent this change as a separate PR here, since I feel that this part is ready to merge regardless of what happens to the other parts of this PR

https://github.com/facebook/jest/pull/13543